python/appledoublekaitai/kaitai_utils.py

```diff
@@ -1,62 +1,200 @@
 import enum
+import importlib
 import inspect
 import json
+import os
 from io import BufferedWriter
 from typing import Any, BinaryIO, Dict, Generator, List, Type
 
-import kaitaistruct
-from kaitaistruct import KaitaiStruct
+import hansken_extraction_plugin
+import yaml
+from hansken_extraction_plugin.api.extraction_trace import ExtractionTrace
+from hansken_extraction_plugin.api.transformation import Range, RangedTransformation
 from json_stream import streamable_dict, streamable_list
+from kaitaistruct import KaitaiStruct
 
 
-def write_to_json(data_binary: BinaryIO, writer: BufferedWriter, class_type: Type[KaitaiStruct]):
+def write_kaitai_to_trace(trace: ExtractionTrace, max_byte_array_length: int):
     """
-    Writes a binary form of JSON string into a BufferedWriter
-
-    @param data_binary: binary data containing the file content
-    @param writer: bufferedWriter to write the binary form of the JSON string to
-    @param class_type: class that contains the parsing to a KaiTai struct
-    @return: JSON string representing contents of data object
+    This method creates traces from a kaitai object tree. It results in a trace containing a JSON representation of the
+    object tree and potential child traces containing byte arrays.
+    @param trace: ExtractionTrace provided to the plugin. Child traces are added to this.
+    @param max_byte_array_length: byte arrays longer than max_byte_array_length are stored as child traces, shorter ones
+    are put into the resulting JSON file.
     """
-    writer.write(bytes(to_json_string(data_binary, class_type), "utf-8"))
 
+    with trace.open(data_type='text', mode='wb') as writer, trace.open() as data:
+        kaitaiclass = _get_kaitai_class()
+        kaitai_to_trace_writer = _KaitaiToTraceWriter(writer, trace, max_byte_array_length)
+        kaitai_to_trace_writer.write_to_trace(data, kaitaiclass)
 
-def to_json_string(data_binary: BinaryIO, class_type: Type[KaitaiStruct]) -> str:
-    """
-    Parses a binary data object to a JSON string
 
-    @param data_binary: binary data containing the file content
-    @param class_type: class that contains the parsing to a KaiTai struct
-    @return: JSON string representing contents of data object
-    """
-    parsed_kaitai_struct = class_type.from_io(data_binary)
-    return json.dumps(_object_to_dict(parsed_kaitai_struct), indent=2)
+def _get_ksy_file():
+    path = os.path.relpath(os.path.join(os.path.dirname(__file__), 'structs'))
+    # path = 'structs'
```

Comment:
Could this line be removed?

Reply:
Copied the new python file from the template, which fixes this.